primitives/account/src/vesting_contract.rs

```diff
@@ -256,6 +256,130 @@ impl AccountTransactionInteraction for VestingContract {
             },
         ])
     }
+    fn commit_failed_transaction(
+        accounts_tree: &AccountsTrie,
+        db_txn: &mut WriteTransaction,
+        transaction: &Transaction,
+    ) -> Result<AccountInfo, AccountError> {
+        let key = KeyNibbles::from(&transaction.sender);
+
+        let account = accounts_tree
+            .get(db_txn, &key)
+            .ok_or(AccountError::NonExistentAddress {
+                address: transaction.sender.clone(),
+            })?;
+
+        let vesting = match account {
+            Account::Vesting(ref value) => value,
+            _ => {
+                return Err(AccountError::TypeMismatch {
+                    expected: AccountType::Vesting,
+                    got: account.account_type(),
+                })
+            }
+        };
+
+        // Note that in this type of transactions the fee is paid (deducted) from the contract balance
+        let new_balance = Account::balance_sub(account.balance(), transaction.fee)?;
```

Comment:
Do we not have to check `min_cap` here as well as the correctness of the recipients signature?
Both of these checks are performed for `commit_outgoing_transaction`.

Reply:
This is covered by _can_pay_fee_ , the mempool before accepting the transaction needs to call _can_pay_fee_ which would check that the transaction signer is contract owner.
Do you need think it is necessary to also add this check here?